configure.ac

```diff
@@ -405,7 +405,7 @@ if test "x$enable_tls" = "xyes"; then
       # Can I compile and link it?
       AC_TRY_LINK([#include <sys/time.h>
   #include <sys/types.h>
-  #include <openssl/ssl.h>], [ SSL_CTX* ssl_ctx = SSL_CTX_new (SSLv23_server_method()); ],
+  #include <openssl/ssl.h>], [ SSL_CTX* ssl_ctx = SSL_CTX_new(TLS_server_method()); ],
```

Comment:
Is this what OPENSSL_API_COMPAT is for? though this is probably fine. if we want to bump up the version again we'd have to find different function though.

Reply:
Looks like OPENSSL_API_COMPAT is to hide deprecated APIs in new builds. Yes, for the current version bump, this validates both compile/link and the version, but ideally we should check the exact version with "$openssl version" or something alike.

Reply:
looks like assert(OPENSSL_VERSION_NUMBER >= 0x10100000L) might be what we want?

Reply:
+1, let me check that.